### PR TITLE
[BugFix] Avoid using the same host name with another existed fe node

### LINKED ISSUE #14249: When modifying the host of node, it should be forbidden to modify it into an existing host
### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
2. INSERT INTO '....'
3. SELECT  '....'

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by waittttting**
…into an existing host

## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #14249 

## Problem Summary(Required) ：

When modifying the host of a fe node, we should avoid using the same host name with another existed fe node

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [x] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Review by Astralidea — DISMISSED** [review]

**Comment by nshangyiming** [comment]
> When modifying the host of node, it should be forbidden to modify it into an existing host

how about "When modifying the host of a fe node, we should aovoid using the same host name with another existed fe node."

And please refine the title of this pr, with simple title and put the details in the problem summary section.

File: fe/fe-core/src/main/java/com/starrocks/server/NodeMgr.java [hunk 1]
```diff
@@ -722,6 +722,12 @@ public void modifyFrontendHost(ModifyFrontendAddressClause modifyFrontendAddress
             if (preUpdateFe == null) {
                 throw new DdlException(String.format("frontend [%s] not found", toBeModifyHost));
             }
+
+            Frontend existFe = getFeByHost(fqdn);
+            if (null != existFe) {
+                throw new DdlException("frontend use host [" + fqdn + "] already exists ");
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/server/NodeMgr.java:728]: use -> with

**Review by nshangyiming — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/server/NodeMgr.java (hunk 1 above)
Comment (waittttting) [reply fe/fe-core/src/main/java/com/starrocks/server/NodeMgr.java]: done

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 3 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/NodeMgr.java|2|3|66.67%|[728]|

**Review by nshangyiming — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Pull request merged**